### PR TITLE
Pchr 773 custom ui select

### PR DESCRIPTION
**Pull request by danilo-compucorp**
This PR introduces a new theme called "civihr-ui-select" for the "ui-select" directive.
It's based on the "select2" theme, with bootstrap look-and-feel, and it contains styles overrides in order to look properly in civihr, either in "appraisals" (which uses the common bootstrap theme) or tasks and assignments (which for the moment, has it's own version of the theme).

Examples:

* Appraisals (multi select):
appraisals-multiple

* Tasks and Assignments (multi select):
t a-multiple

* Appraisals (single select):
appraisals-single

* Tasks and Assignments (single select):
t a-single

### CODE REVIEW

**Comment by JonScreat** [comment]
All looks good to me. Jon

**Comment by danilo-compucorp** [comment]
@JonScreat I almost forgot... the "user icons" (on the autocomplete results) weren't supposed to be part of this PR.. so I removed those and updated the screenshots.

**Comment by JonScreat** [comment]
okay, got it - looks good

File: org.civicrm.reqangular/src/common/decorators/ui-select/ui-select-choices-custom-scrollbars.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+define([
+  'common/vendor/perfect-scrollbar'
+], function (ps) {
+  'use strict';
+
+  /**
+   * Decorates ui-select-choices directive, in order for it to have custom scrollbars,
+   * using the "perfect-scrollbar" plugin.
+   */
+  return ['$delegate', function ($delegate) {
```
Comment (kkalamarski) [inline org.civicrm.reqangular/src/common/decorators/ui-select/ui-select-choices-custom-scrollbars.js]: To improve readability please separate variable declarations, returns and other statements from the rest of the code with one empty line.

```javascript
return ['$delegate', function ($delegate) {
    var directive = $delegate[0];
    var origCompile = directive.compile;

    directive.compile = function compile() {
      var link = origCompile.apply(this, arguments);

      return function (scope, element) {
        link.apply(this, arguments);

        if (element.closest('.civihr-ui-select').length) {
          ps.initialize(element[0]);
        }
      };
    };

    return $delegate;
  }];
```

**Comment by kkalamarski** [comment]
Good job! When corrected formatting, approved.

**Pull request merged**